lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -46,10 +62,59 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
+        def relevant_included_associations_for(serializer, nested_associations = {})
+          if nested_associations.present?
+            serializer.associations.select do |association|
+              # nested_associations is a hash of:
+              #   key => nested association to include
+              nested_associations.key?(association.name)
+            end
+          else
+            serializer.associations
+          end
+        end
+
+        # add a singular relationship
+        # the options should always belong to the serializer
+        def to_one_relationship_for(serializer, options)
+          if options[:include].present?
+            Attributes.new(serializer, options).serializable_hash(options)
+          else
+            serialized_or_virtual_of(serializer, options)
+          end
+        end
+
+        # add a many relationship
+        def to_many_relationship_for(serializer, options)
+          serializer.map do |item_serializer|
+            if options[:include].present?
+              Attributes.new(item_serializer, options).serializable_hash(options)
+            else
+              resource_object_for(item_serializer, options)
+            end
+          end
+        end
+
         # no-op: Attributes adapter does not include meta data, because it does not support root.
         def include_meta(json)
```

Comment:
guess :) `include_meta` includes meta. In the super class

```ruby
      def include_meta(json)
        json[meta_key] = meta if meta
        json
      end
```

So, really, if we do a better job of making this a no-op by testing for an attribute, we wouldn't need it in the Attributes adapter.